src/components/pdf-upload/PdfUpload.tsx

```diff
@@ -0,0 +1,485 @@
+      file,
+      status: "pending",
+      progress: 0,
+      uploadUrl: null,
+      uploadId: null,
+      error: null,
+      result: null,
+    }));
+
+    setFileStates((prev) => [...prev, ...newFileStates]);
+    onUploadStart?.(selectedFiles);
+  };
+
+  const updateFileState = (
+    fileIndex: number,
+    updates: Partial<FileUploadState>
+  ) => {
+    setFileStates((prev) =>
+      prev.map((state, index) =>
+        index === fileIndex ? { ...state, ...updates } : state
+      )
+    );
+  };
+
+  const convertFileToBase64 = (file: File): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = () => {
+        const result = reader.result as string;
+        // Remove the data:application/pdf;base64, prefix
+        const base64 = result.split(",")[1];
+        resolve(base64);
+      };
+      reader.onerror = (error) => reject(error);
+    });
+  };
+
+  const uploadSingleFile = async (fileIndex: number) => {
+    const fileState = fileStates[fileIndex];
+    if (!fileState || fileState.status !== "pending") return;
+
+    if (!adminSecret) {
+      const errorMessage = "Admin secret required for PDF upload";
+      updateFileState(fileIndex, {
+        status: "error",
+        error: errorMessage,
+      });
+      onFileUploadError?.(fileState.file, errorMessage);
+      return;
+    }
+
+    // Choose upload method based on file size
+    const fileSizeMB = fileState.file.size / 1024 / 1024;
+    const usePresignedUrl = fileSizeMB > PDF_CONFIG.PRESIGNED_URL_THRESHOLD_MB;
+
+    if (usePresignedUrl) {
+      return await uploadViaPresignedUrl(fileIndex);
+    }
+    return await uploadViaBase64(fileIndex);
+  };
+
+  const uploadViaBase64 = async (fileIndex: number) => {
+    const fileState = fileStates[fileIndex];
+
+    try {
+      // Step 1: Convert file to base64
+      updateFileState(fileIndex, { status: "generating-url", progress: 10 });
+      const base64Data = await convertFileToBase64(fileState.file);
+
+      // Step 2: Upload using the uploadPdfFile tool
+      updateFileState(fileIndex, { status: "uploading", progress: 50 });
+
+      const result = await invokeTool("uploadPdfFile", {
+        filename: fileState.file.name,
+        fileData: base64Data,
+        adminSecret: adminSecret,
+      });
+
+      updateFileState(fileIndex, {
+        status: "completed",
+        result: result || "Upload completed",
+        progress: 100,
+      });
+
+      onFileUploadComplete?.(fileState.file, result);
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : "Upload failed";
+      updateFileState(fileIndex, {
+        status: "error",
+        error: errorMessage,
+      });
+      onFileUploadError?.(fileState.file, errorMessage);
+    }
+  };
+
+  const uploadViaPresignedUrl = async (fileIndex: number) => {
```

Comment:
@aniham here's the presigned url upload. Unless I misunderstand your question about moving away from it.